src/sidebar/activateRecipe/RequireRecipe.tsx

```diff
@@ -57,7 +60,24 @@ function requiresUserConfiguration(
     authOptions
   );
 
-  const hasRecipeOptions = !isEmpty(recipe.options?.schema?.properties);
+  const recipeOptions = recipe.options?.schema?.properties ?? {};
+
+  // Options that allow auto-activation:
+  // - Database fields with format "preview"
+  const needsOptionsInputs =
+    !isEmpty(recipeOptions) &&
+    Object.values(recipeOptions).some((optionSchema) => {
+      if (typeof optionSchema === "boolean") {
+        return false;
+      }
+
+      // Exclude preview database fields
+      if (isDatabaseField(optionSchema) && optionSchema.format === "preview") {
+        return false;
+      }
+
+      return true;
+    });
```

Comment:
This reads a little easier IMO

```suggestion
  const recipeOptions = recipe.options?.schema?.properties;

  // Options that allow auto-activation:
  // - Database fields with format "preview"
  const needsOptionsInputs =
    recipeOptions != null &&
    Object.values(recipeOptions).some((optionSchema) => 
      typeof optionSchema !== "boolean" && 
      isDatabaseField(optionSchema) && 
      optionSchema.format === "preview"
    );
```

Reply:
I wrote it like this because I anticipate adding more clauses in the future for other types of things that we want to auto-activate. I can add a comment explaining the true/false return values a little better though.

Also, your suggestion reverses the logic, we want "preview" to make it return false there, not true.

Reply:
> Also, your suggestion reverses the logic, we want "preview" to make it return false there, not true.

That's why I said to try it lol.